ciao-controller/internal/datastore/datastore.go

```diff
@@ -2280,3 +2257,61 @@ func (ds *Datastore) UnMapExternalIP(address string) error {
 
 	return nil
 }
+
+func (ds *Datastore) generateCNCIWorkload() {
+	// generate the CNCI workload.
+	config := `---
+#cloud-config
+users:
+  - name: cloud-admin
+    gecos: CIAO Cloud Admin
+    lock-passwd: false
+    passwd: $6$rounds=4096$w9I3hR4g/hu$AnYjaC2DfznbPSG3vxsgtgAS4mJwWBkcR74Y/KHNB5OsfAlA4gpU5j6CHWMOkkt9j.9d7OYJXJ4icXHzKXTAO.
+    sudo: ALL=(ALL) NOPASSWD:ALL
+...
+`
+	cpus := payloads.RequestedResource{
+		Type:      payloads.VCPUs,
+		Value:     4,
+		Mandatory: false,
+	}
+
+	mem := payloads.RequestedResource{
+		Type:      payloads.MemMB,
+		Value:     128,
```

Comment:
>  how do you feel about repurposing the "singlemachine" parameter to add cnci configuration for singleVM? 

I don't really understand what is meant by the "singlemachine" parameter.  I do understand your concern about the PR getting too big though.   So if you have a way, even a hacky way, to ensure both single vm and normal ciao clusters work as before after this PR is committed, that's good enough for me.